packages/emptystate/src/react/index.js

```diff
@@ -110,55 +108,35 @@ Illustration.propTypes = {
   name: PropTypes.oneOf(Object.values(Illustration.names))
 }
 
-class EmptyState extends React.PureComponent {
-  constructor(props) {
-    super(props)
-
-    this.renderSmallIfElementLessThan = 450
-  }
-
-  renderContent(ctx) {
-    return (
-      <Context.Provider value={ctx}>
-        <div
-          {...styles.emptyState(this.props, ctx)}
-          {...filterReactProps(this.props)}
-        >
-          {this.props.illustration}
-          {this.props.heading}
-          {this.props.caption}
-          {this.props.actions}
-        </div>
-      </Context.Provider>
-    )
-  }
-
-  render() {
-    const { size, themeName } = this.props
-
-    const hasSizeOverride = !!size
-
-    if (hasSizeOverride) {
-      const ctx = { hasRenderedOnce: true, size, themeName }
-      return this.renderContent(ctx)
-    }
-
-    return (
-      <ResizeObserver observeHeight={false}>
-        {({ width }) => {
-          const hasRenderedOnce = !!width
-          const size =
-            hasRenderedOnce && width <= this.renderSmallIfElementLessThan
-              ? vars.sizes.small
-              : vars.sizes.large
-
-          const ctx = { hasRenderedOnce, size, themeName }
-          return this.renderContent(ctx)
-        }}
-      </ResizeObserver>
-    )
-  }
-}
+const EmptyState = withTheme(function EmptyState(props) {
+  const [hasRenderedOnce, setHasRenderedOnce] = useState(false)
+  const [ref, { width }] = useResizeObserver()
```

Comment:
kind of an odd api to return an array with an object that is then destructured.  Why not just return an object and then destructure ref out of it as well?  Seems it would simplify the data structure.

Reply:
yeah i can see that. i'll update